### PR TITLE
Speed unit testing.

### PR DESCRIPTION
**Pull request by qingqing01**
1. Refine TensorSetElement.
2. Refine CUDA kernel.
3. Reduce testing size in test_affine_channel

On my machine:
120 s -> 30s

### CODE REVIEW

File: paddle/fluid/pybind/tensor_py.h [hunk 1]
```diff
@@ -141,20 +141,22 @@ T TensorGetElement(const framework::Tensor &self, size_t offset) {
...
 template <typename T>
 void TensorSetElement(framework::Tensor *self, size_t offset, T elem) {
   if (platform::is_gpu_place(self->place())) {
-    framework::Tensor dst;
-    framework::TensorCopySync(*self, platform::CPUPlace(), &dst);
-    dst.mutable_data<T>(platform::CPUPlace())[offset] = elem;
-    framework::TensorCopySync(dst, self->place(), self);
+    auto p = boost::get<platform::CUDAPlace>(self->place());
+    T *a = self->mutable_data<T>(p);
+    paddle::memory::Copy(p, a + offset, platform::CPUPlace(), &elem, sizeof(T),
```
Comment (wopeizl) [inline paddle/fluid/pybind/tensor_py.h:165]: 是不是需要加offset >= len(data) || offset < -len(data) 的检查？

Comment (qingqing01) [reply paddle/fluid/pybind/tensor_py.h:165]: Add check. Thansk!

**Review by wopeizl — DISMISSED** [review]

**Review by chengduoZH — APPROVED** [review]
LGTM

**Review by sneaxiy — APPROVED** [review]
LGTM.

**Pull request merged**